Update tpl_shopping_cart_default.php

retched opened the pull request:
Hides the Update Cell in the Shopping Cart unless the option `SHOW_SHOPPING_CART_UPDATE` is specifically set to `Next to each Qty Box` or `Both Next to each Qty Box and Below all Products`.

retched commented:
Running the shopping card through an evaluation tool suggested that to comply with WCAG standard 1.3.1 a `<TH>` tag shouldn't be empty or contain images with no alt text.

So I hide that column that would contain the update button UNLESS it is explicitly set in the Admin dashboard. As well as change the last column, the total column, to a regular `<TD>` instead of a `<TH>`.

retched commented:
It was a suggestion that <th>'s shouldn't be empty, even with a non-breaking space, for accessibility. To quote the guideline I found:

https://webaim.org/techniques/tables/data#th
Table headers should never be empty. This is particularly of concern for the top-left cell of some tables

I'm not sure how stringent that is though.

========
Respectfully submitted,


Paul L. Williams III
________________________________
From: lat9 ***@***.***>
Sent: Friday, June 28, 2024 11:11 AM
To: lat9/ZCA-Bootstrap-Template ***@***.***>
Cc: Paul Williams ***@***.***>; Author ***@***.***>
Subject: Re: [lat9/ZCA-Bootstrap-Template] Update tpl_shopping_cart_default.php (PR #403)


Thanks for this. I'll review later this weekend. Why the change to td instead of th?

—
Reply to this email directly, view it on GitHub<https://github.com/lat9/ZCA-Bootstrap-Template/pull/403#issuecomment-2197150733>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAL75NFPTD6U5IG7RIOXL53ZJV4KPAVCNFSM6AAAAABKB7Z4RSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOJXGE2TANZTGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

drbyte commented:
One way we handle the "no empty header cell" elsewhere in Zen Cart is to simply give it a header of "Action". 
And often make that text aria-only.

retched commented:
I didn't even consider that as a possibility. I can make a similar change in another commit.

retched commented:
I reverted the empty cells to a `<th>` but gave each column an aria-label header with an empty cell. I also added two language definitions to `lang.zca_bootstrap_common.php` for this purpose.

lat9 commented on includes/templates/bootstrap/templates/tpl_shopping_cart_default.php:
```diff
@@ -70,11 +70,11 @@
             <tr>
                 <th scope="col" id="cartTableDisplay-qtyHeading"><?php echo TABLE_HEADING_QUANTITY; ?></th>
-                <th scope="col" class="d-none d-sm-table-cell" id="cartTableDisplay-qtyUpdateHeading">&nbsp;</th>
+                <?php if (SHOW_SHOPPING_CART_UPDATE == 1 || SHOW_SHOPPING_CART_UPDATE == 3) { ?><th scope="col" class="d-none d-sm-table-cell" id="cartTableDisplay-qtyUpdateHeading"><span aria-label="<?php echo TEXT_CART_ARIA_HEADING_UPDATE_COLUMN; ?>">&nbsp;</span></th> <?php } ?>
```
I'm bit bit more persnickety and don't like the 'inline' PHP conditionals.  Please indent the `<php` conditional to line up with its parent and render the HTML portion separately (also indented to match the other `th` elements).

lat9 commented on includes/templates/bootstrap/templates/tpl_shopping_cart_default.php:
```diff
@@ -108,7 +108,7 @@
                     </div>
                 </td>
-                <td class="qtyUpdateCell text-center d-none d-sm-table-cell"><?php echo (!empty($product['buttonUpdate'])) ? $product['buttonUpdate'] : ''; ?></td>
+                <?php if (SHOW_SHOPPING_CART_UPDATE == 1 || SHOW_SHOPPING_CART_UPDATE == 3) { ?><td class="qtyUpdateCell text-center d-none d-sm-table-cell"><?php echo (!empty($product['buttonUpdate'])) ? $product['buttonUpdate'] : ''; ?></td><?php } ?>
```
Ditto for the `<td>`.

lat9 reviewed (changes requested):
Resolve those couple of indentation 'issues' and it's good to go.

retched commented:
Should be done. Also went ahead and adjusted the quantity field to render the HTML aligned within the `<td>` itself too.
![image](https://github.com/lat9/ZCA-Bootstrap-Template/assets/1572532/551b5dc9-e1ef-4efa-ad4c-fa18651909ca)

retched commented:
Squashed the three commits to make things easier to follow.

lat9 commented:
Tie this PR to the associated issue, #410.
